base/applications/rapps/installed.cpp

```diff
@@ -71,8 +71,40 @@ void CInstalledApplicationInfo::EnsureDetailsLoaded()
         }
         GetApplicationRegString(L"InstallLocation", szInstallLocation);
         GetApplicationRegString(L"InstallSource", szInstallSource);
-        GetApplicationRegString(L"UninstallString", szUninstallString);
-        GetApplicationRegString(L"ModifyPath",szModifyPath);
+        DWORD dwWindowsInstaller = 0;
+        if (GetApplicationRegDword(L"WindowsInstaller", &dwWindowsInstaller) && dwWindowsInstaller)
+        {
+            // MSI has the same info in Uninstall / modify, so manually build it
+            szUninstallString.Format(L"msiexec /x%s", m_szKeyName.GetString());
+        }
+        else
+        {
+            GetApplicationRegString(L"UninstallString", szUninstallString);
+        }
+        DWORD dwNoModify = 0;
+        if (!GetApplicationRegDword(L"NoModify", &dwNoModify))
+        {
+            CStringW Tmp;
+            if (GetApplicationRegString(L"NoModify", Tmp))
+            {
+                dwNoModify = Tmp.GetLength() > 0 ? (Tmp[0] == '1') : 0;
```

Comment:
What if the string is "2" or something else? If it's considered to be like a BOOL variable, usually it's FALSE when it's zero, and TRUE when it's anything else.

Reply:
This is already a fallback, so let's not write a fallback on top of a fallback :)